main.go

```diff
@@ -84,23 +95,51 @@ func main() {
 		glog.Fatal(err)
 	}
 
-	var metricConfig *util.MetricConfig
-	if *enableMetrics {
-		metricConfig = &util.MetricConfig{
-			MetricsEndpoint: *metricsEndpoint,
-			MetricsPort:     *metricsPort,
-			MetricsPrefix:   *metricsPrefix,
-			MetricsLabels:   metricsLabels,
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+
+	stopCh := make(chan struct{}, 1)
+	startCh := make(chan struct{}, 1)
+
+	if *enableLeaderElection {
+		hostname, err := os.Hostname()
+		if err != nil {
+			glog.Fatal(err)
+		}
+		resourceLock, err := resourcelock.New(resourcelock.ConfigMapsResourceLock, *leaderElectionLockNamespace, *leaderElectionLockName, kubeClient.CoreV1(), resourcelock.ResourceLockConfig{
```

Comment:
Do we need to remove the lock upon exiting?

Reply:
I don't think so, that could cause an unnecessary re-election when a non-leader pod is restarted. And when the leader loses its lease, it restarts, so that could cause some thrashing.

Reply:
Got it.